pkg/config/store/yaml/store.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/afero"
+	"sigs.k8s.io/yaml"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+)
+
+const (
+	// DefaultPath is the default path for the configuration file
+	DefaultPath = "PROJECT"
+)
+
+// yamlStore implements store.Store using a YAML file as the storage backend
+// The key is translated into the YAML file path
+type yamlStore struct {
```

Comment:
I do not think that the structure/object should have in its name the type of file used by it. 
Why need yamlStore and Store? Why the yaml stuff cannot be attributes of Store?

Reply:
`Store` is an interface so that in a future, we can use other persistence options. For example, we may want to store the project configuration in a database, or in JSON, or maybe base64 encoded so that users do not modify it, ... In this case, we would create a `jsonStore` or equivalent struct that would implement `Store` and we would allow the cli to choose between them somehow (probably a `WithStoreBackend` or similar option to `NewCLI` so that each third-party command can choose which one they use).

Reply:
json/yaml/txt shows values of the attribute format of some structure such as  PluginConfigData.

Reply:
I do not see it as a rfe, however, it is fine for now.